arangod/Scheduler/Scheduler.cpp

```diff
@@ -254,25 +306,30 @@ bool Scheduler::queue(RequestPriority prio,
 
   switch (prio) {
     case RequestPriority::HIGH:
-      if (0 < _fifoSize[0]) {
-        ok = pushToFifo(static_cast<int>(prio), callback);
+      if (0 < _fifoSize[FIFO1]) {
+        ok = pushToFifo(FIFO1, callback, false);
       } else if (canPostDirectly()) {
-        post(callback);
+        post(callback, false);
       } else {
-        ok = pushToFifo(static_cast<int>(prio), callback);
+        ok = pushToFifo(FIFO1, callback, false);
```

Comment:
`pushToFifo(FIFO1)` appears in the `if` and the `else` branches... Is this desired?